src/v/pandaproxy/schema_registry/sharded_store.h

```diff
@@ -118,6 +118,10 @@ class sharded_store {
     ss::future<compatibility_level>
     get_compatibility(subject sub, default_to_global fallback);
 
+    ///\brief Check the compatibility level for a subject w/o fallback.
+    /// If none exists, the underlying store returns subject_not_found
+    ss::future<compatibility_level> check_compatibility(subject sub);
```

Comment:
I agree, perhaps the check could be done within `seq_writer::do_delete_config` and that method can return a `result<optional<bool>>`?

Reply:
> I agree, perhaps the check could be done within seq_writer::do_delete_config and that method can return a result<optional<bool>>?

Specially since it seems that `check_compatible` is only used for the delete use-case.

Reply:
Removed the extraneous function. Since the `seq_writer` implementations need to return `bool`, we'll still need to call `get_compatibility` from the handler (I think) in order to conform to the API. Am I reading that right?

Reply:
Oh, I see `delete_subject_permanent_inner` does something similar to what you're describing @NyaliaLui. Makes sense.

Reply:
Played around with this a bit, and I think performing the check up in the handler is more or less in line with surrounding code (and less of a pain). Happy to revisit if that interpretation sounds wrong.